Remove all of Lukas Oberhuber's misguided hacking of widget focus.
diff --git a/gdk/quartz/gdkwindow-quartz.c b/gdk/quartz/gdkwindow-quartz.c
index 1405a93c22..672103f7bd 100644
--- a/gdk/quartz/gdkwindow-quartz.c
+++ b/gdk/quartz/gdkwindow-quartz.c
@@ -791,12 +791,31 @@ _gdk_quartz_window_did_become_main (GdkWindow *window)
 void
 _gdk_quartz_window_did_resign_main (GdkWindow *window)
 {
-  /* Don't try to make another window key when one resigns main.
-   * macOS handles this automatically, and GTK's attempt to manage focus
-   * here causes focus fighting between windows.
-   * See: https://gitlab.gnome.org/GNOME/gimp/-/issues/14901
-   *      https://gitlab.gnome.org/GNOME/gimp/-/issues/15480
-   */
+  GdkWindow *new_window = NULL;
+
+  if (main_window_stack)
+    new_window = main_window_stack->data;
+  else
+    {
+      GList *toplevels;
+
+      toplevels = gdk_screen_get_toplevel_windows (gdk_screen_get_default ());
+      if (toplevels)
+        new_window = toplevels->data;
+      g_list_free (toplevels);
+    }
+
+  if (new_window &&
+      new_window != window &&
+      GDK_WINDOW_IS_MAPPED (new_window) &&
+      WINDOW_IS_TOPLEVEL (new_window))
+    {
+      GdkWindowImplQuartz *impl = gdk_window_get_quartz_impl (new_window);
+
+      if (impl)
+        [impl->toplevel makeKeyAndOrderFront:impl->toplevel];
+    }
+
   clear_toplevel_order ();
 }

@@ -1164,15 +1164,8 @@ gdk_window_quartz_show (GdkWindow *window, gboolean already_mapped)
       make_key = (window->accept_focus && focus_on_map &&
                   window->window_type != GDK_WINDOW_TEMP);
 
-      /* If window is already mapped and visible, skip showAndMakeKey to avoid
-       * spurious resignMain/becomeMain notifications that cause focus fighting.
-       * See https://gitlab.gnome.org/GNOME/gimp/-/issues/14901
-       */
-      if (!GDK_WINDOW_IS_MAPPED (window) || ![impl->toplevel isVisible])
-        {
-          [(GdkQuartzNSWindow*)impl->toplevel showAndMakeKey:make_key];
-          clear_toplevel_order ();
-        }
+      [(GdkQuartzNSWindow*)impl->toplevel showAndMakeKey:make_key];
+      clear_toplevel_order ();
 
       _gdk_quartz_events_send_map_event (window);
     }
-- 
2.50.1 (Apple Git-155)

